src/Authenticator.ts

```diff
@@ -3,63 +3,117 @@ import Mustache = require('mustache');
 import readPkg = require('read-pkg');
 import path = require('upath');
 
-const Zip = require('node-zip');                   // tslint:disable-line:no-var-requires  no type definitions available
+const Zip = require('node-zip'); // tslint:disable-line:no-var-requires  no type definitions available
 import { coerce, SemVer } from 'semver';
 import { ensure, isArray, isGreaterThan, isString, property } from 'tiny-types';
 
 export class Authenticator {
-    /**
-     * @param {string} username
-     * @param {string} password
-     * @param {string[]} permissions
-     *  See https://developer.chrome.com/extensions/declare_permissions
-     */
-    static for(username: string, password: string, permissions: string[] = ['<all_urls>']): Authenticator {
-        return new Authenticator(username, password, permissions);
-    }
+  /**
+   * @param {string} username
+   * @param {string} password
+   * @param {string[]} permissions
+   *  See https://developer.chrome.com/extensions/declare_permissions
+   */
+  static for(
+    username: string,
+    password: string,
+    permissions: string[] = ['<all_urls>']
+  ): Authenticator {
+    return new Authenticator(username, password, permissions);
+  }
 
-    asBase64(): string {
-        return this.extension().generate({ base64: true, compression: 'DEFLATE' });
-    }
+  asBase64(): string {
+    return this.extension().generate({ base64: true, compression: 'DEFLATE' });
+  }
+
+  asFileAt(_path: string): string {
```

Comment:
I'd avoid the `_` prefix; we could give this variable a more meaningful name and this way avoid the conflict with the `upath` module aliased as `path`. How about calling it `pathToDestinationDirectory`. Yeah, it's wordy, but way whoever looks at the code will immediately understand what it's for.

Reply:
changed as per suggested